examples/tti/tti_operators.py

```diff
@@ -41,70 +41,94 @@ def ForwardOperator(model, u, v, src, rec, data, time_order=2,
+                                dim=z, side=right, order=spc_brd,
+                                matvec=transpose))
         Gzz = (first_derivative(Gzr * ang1 * ang2,
-                                dim=x, side=centered, order=spc_brd) +
+                                dim=x, side=centered, order=spc_brd,
+                                matvec=transpose) +
                first_derivative(Gzr * ang1 * ang3,
-                                dim=y, side=left, order=spc_brd) +
+                                dim=y, side=right, order=spc_brd,
+                                matvec=transpose) +
                first_derivative(Gzr * ang0,
-                                dim=z, side=left, order=spc_brd))
+                                dim=z, side=right, order=spc_brd,
+                                matvec=transpose))
         Gxp2 = (ang0 * ang2 * u.dxr + ang0 * ang3 * u.dy - ang1 * u.dz)
         Gzr2 = (ang1 * ang2 * v.dxr + ang1 * ang3 * v.dy + ang0 * v.dz)
         Gxx2 = (first_derivative(Gxp2 * ang0 * ang2,
-                                 dim=x, side=left, order=spc_brd) +
+                                 dim=x, side=right, order=spc_brd,
+                                 matvec=transpose) +
                 first_derivative(Gxp2 * ang0 * ang3,
-                                 dim=y, side=centered, order=spc_brd) -
+                                 dim=y, side=centered, order=spc_brd,
+                                 matvec=transpose) -
                 first_derivative(Gxp2 * ang1,
-                                 dim=z, side=centered, order=spc_brd))
+                                 dim=z, side=centered, order=spc_brd,
+                                 matvec=transpose))
         Gzz2 = (first_derivative(Gzr2 * ang1 * ang2,
-                                 dim=x, side=left, order=spc_brd) +
+                                 dim=x, side=right, order=spc_brd,
+                                 matvec=transpose) +
                 first_derivative(Gzr2 * ang1 * ang3,
-                                 dim=y, side=centered, order=spc_brd) +
+                                 dim=y, side=centered, order=spc_brd,
+                                 matvec=transpose) +
                 first_derivative(Gzr2 * ang0,
-                                 dim=z, side=centered, order=spc_brd))
-        Hp = (.5*Gxx + .5*Gxx2 + .5 * Gyy + .5*Gyy2)
-        Hzr = (.5*Gzz + .5 * Gzz2)
+                                 dim=z, side=centered, order=spc_brd,
+                                 matvec=transpose))
+        Hp = -(.5*Gxx + .5*Gxx2 + .5 * Gyy + .5*Gyy2)
+        Hzr = -(.5*Gzz + .5 * Gzz2)
 
     else:
         Gx1p = (ang0 * u.dxr - ang1 * u.dy)
         Gz1r = (ang1 * v.dxr + ang0 * v.dy)
         Gxx1 = (first_derivative(Gx1p * ang0, dim=x,
-                                 side=left, order=spc_brd) -
+                                 side=right, order=spc_brd,
+                                 matvec=transpose) -
```

Comment:
the expression of the derivative is Dx^T.Dx . It used to be done "by hand" by knowing what the transpose is supposed to be, but this way it is actually readable and implement what you would write on paper.